File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        // for(int i = 0; i < nums.Length - 1; i++)
        // {
        //     for(int j = 1; j < nums.Length; j++)
        //     {
        //         if (nums[i] + nums[j] == target)
        //         {
        //             return new int[] {i, j};
        //         }
        //     }
        // }
        // return null;

        var hash = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++)
        {
            var lookUp = target - nums[i];
            if(!hash.ContainsKey(lookUp))
            {
                hash[nums[i]] = i;
            }
            else
            {
                return new int[]{hash[lookUp], i};
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        j = len(numbers) - 1

        while i < j:
            comb = numbers[i] + numbers[j]

            if target < comb:
                j-=1
            elif target > comb:
                i+=1
            else:
                return [i+1, j+1]

        return []